### PR TITLE
Change audience from Option[String] to Option[Set[String]]

### PR DESCRIPTION
**Pull request by stanch**
According to https://tools.ietf.org/html/rfc7519#section-4.1.3, the "aud" claim is normally an array of strings, but may be a single string.

I am not sure what is the best approach to add regression tests to the project, but the current tests pass (when running `testAll`). Maybe someone could suggest the way to organize the new test code, or just pick it up from here :)

### CODE REVIEW

File: json/json4s-common/src/main/scala/JwtJson4sCommon.scala [hunk 1]
```diff
@@ -2,10 +2,11 @@ package pdi.jwt
 
 import org.json4s._
 import org.json4s.JsonDSL.WithBigDecimal._
-
-import pdi.jwt.exceptions.{JwtNonStringException, JwtNonNumberException}
+import pdi.jwt.exceptions.{JwtNonNumberException, JwtNonStringException, JwtNonStringSetOrStringException}
 
 trait JwtJson4sCommon extends JwtJsonCommon[JObject] {
+  protected implicit def formats: Formats
```
Comment (pauldijou) [inline json/json4s-common/src/main/scala/JwtJson4sCommon.scala:8]: Don't know much about Json4s, but why is that needed?

Comment (stanch) [inline json/json4s-common/src/main/scala/JwtJson4sCommon.scala:8]: This is for `.extract[Set[String]]`, since `extract` needs `Formats`.
